### PR TITLE
Include hint svidstore

### PR DESCRIPTION
**Pull request by guilhermocc**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
`svidstore` plugins.

**Description of change**
Include hint field in the SVIDS stored by `svidstore` plugins.

**Which issue this PR fixes**
Ongoing work for https://github.com/spiffe/spire/issues/2588

**Manual test results**

**Aws svidstore plugin**
- Creating entries with and without hint
image
image
- Update entries
image
image
- Delete entries
image
image

**GCP svidstore plugin**
- Creating entries with and without hint
image
image
- Update entries
image
image
- Delete entries
image

### CODE REVIEW

File: pkg/agent/plugin/svidstore/svidstore.go [hunk 1]
```diff
@@ -43,4 +43,7 @@ type SVID struct {
 
 	// Expiration timestamp
 	ExpiresAt time.Time
+
+	// Hint is used for indicating the svid usage
+	Hint string
```
Comment (MarcosDY) [inline pkg/agent/plugin/svidstore/svidstore.go:48]: What do you think about move hint to be a property of a secret (like spire property we are already storing) 
that will allow users to see hints of a secret before consuming it (that will require to get secret object and parse) 
That will allow uses to filter by hint too

File: pkg/agent/plugin/svidstore/utils.go [hunk 2]
```diff
@@ -50,12 +52,18 @@ func SecretFromProto(req *svidstorev1.PutX509SVIDRequest) (*Data, error) {
 		return nil, fmt.Errorf("failed to parse key: %w", err)
 	}
 
+	hint, err := getHintFromMetadata(req.Metadata)
```
[the remark this answers is not in the document]
Comment (maxlambrecht) [reply pkg/agent/plugin/svidstore/utils.go:55]: I think `Hint` could be part of the `svidstorev1.X509SVID` object.

[the remark this answers is not in the document]
Comment (guilhermocc) [reply pkg/agent/plugin/svidstore/utils.go:55]: Totaly makes sense, I'm going to update my PR with those changes, thanks for the review guys!

**Comment by guilhermocc** [comment]
We can't store hint as metadata in the current plugin due to specific validation that GCP and AWS have. The maintainers have agreed on not storing the hint on the secret object at the moment, so I'm closing this PR.

**Pull request closed without merging**